madmom/evaluation/key.py

```diff
@@ -184,20 +307,45 @@ def __init__(self, eval_objects, name=None):
         self.name = name or 'mean for {:d} files'.format(len(eval_objects))
 
         n = len(eval_objects)
-        c = Counter(e.error_category for e in eval_objects)
-
-        self.correct = float(c['correct']) / n
-        self.fifth = float(c['fifth']) / n
-        self.relative = float(c['relative']) / n
-        self.parallel = float(c['parallel']) / n
-        self.other = float(c['other']) / n
-        self.weighted = sum(e.score for e in eval_objects) / n
+        if n > 0:
+            c = Counter(e.error_category for e in eval_objects)
+
+            self.correct = float(c['correct']) / n
+            self.fifth = float(c['fifth']) / n
+            self.relative = float(c['relative']) / n
+            self.relative_of_fifth = float(c['relative_of_fifth']) / n
+            self.parallel = float(c['parallel']) / n
+            self.other = float(c['other']) / n
+            # Check that all the error_scores in the eval_objects are the same
+            # (otherwise a weighted result would be hard to interpret)
+            if self._check_error_scores(eval_objects):
+                self.weighted = sum(e.score for e in eval_objects) / n
+            else:
+                raise ValueError('Different error_scores found in the '
+                                 'KeyEvaluation objects.')
+        else:
+            raise ValueError('The list of evaluations is empty.')
+
+    def _check_error_scores(self, eval_objects):
+        all_the_same = True
+        indScores = 0
+        while (indScores < len(eval_objects) - 1) and all_the_same:
+            if eval_objects[indScores].error_scores \
+                    != eval_objects[indScores + 1].error_scores:
+                all_the_same = False
+                break
+            else:
+                indScores += 1
+        return all_the_same
 
     def tostring(self, **kwargs):
         return ('{}\n  Weighted: {:.3f}  Correct: {:.3f}  Fifth: {:.3f}  '
-                'Relative: {:.3f}  Parallel: {:.3f}  Other: {:.3f}'.format(
-                    self.name, self.weighted, self.correct, self.fifth,
-                    self.relative, self.parallel, self.other))
+                'Relative: {:.3f}  Relative of Fifth: {:.3f} '
```

Comment:
We should only report "Relative of Fifth" here if it is actually present in all the evaluation objects.

Reply:
OK